silkworm/rpc/ethdb/kv/backend_providers.hpp

```diff
@@ -33,4 +33,16 @@ inline db::chain::BlockNumberFromTxnHashProvider block_number_from_txn_hash_prov
     };
 }
 
+inline db::chain::BlockNumberFromBlockHashProvider block_number_from_block_hash_provider(ethbackend::BackEnd* backend) {
+    return [backend](HashAsSpan hash) -> Task<BlockNum> {
+        co_return co_await backend->get_block_number_from_hash(hash);
+    };
+}
+
+inline db::chain::BlockHashFromBlockNumberProvider block_hash_from_block_number_provider(ethbackend::BackEnd* backend) {
+    return [backend](BlockNum number) -> Task<evmc::bytes32> {
+        co_return co_await backend->get_block_hash_from_block_number(number);
+    };
```

Comment:
`BlockHashFromBlockNumberProvider`, `block_hash_from_block_number_provider`, `get_block_hash_from_block_number` are indeed providing the _canonical_ block hash for the given block number.
I'm wondering if this could be better captured in these names without making them unnecessarily long and cumbersome